src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  chakra,
+  useMultiStyleConfig,
+  // useColorModeValue,
+} from "@chakra-ui/react";
+import List from "../List/List";
+import Link from "../Link/Link";
+import Icon, { IconSizes } from "../Icons/Icon";
+import { LayoutTypes } from "../../helpers/types";
+import { socialMediaDataMap } from "./SocialMediaDataMap";
+import React, { forwardRef } from "react";
+
+export const borderTypeArray = ["none", "circular", "straight"] as const;
+export type BorderType = typeof borderTypeArray[number];
+
+export const colorTypeArray = ["link", "textDefault", "textInverse"] as const;
+export type ColorType = typeof colorTypeArray[number];
+
+export const sizeTypeArray = ["small", "medium", "large"] as const;
+export type SizeType = typeof sizeTypeArray[number];
+
+// @todo The below export fails TS, for reasons I do not yet understand. But it would be useful so that we don't
```

Comment:
I think all you need to do is remove the "as const". "as const" works for when the array is defined and compiled but with the map function, it is dynamically generating the array when it compiles so it gets confused. It's fine to not set this dynamic array "as const" since, with the map function, we're saying it can change.

Reply:
I agree it's nice to define the set of values in one location.